packages/vanilla-renderers/src/complex/categorization/CategorizationList.tsx

```diff
@@ -57,22 +56,20 @@ export const CategorizationList = ({
 }: CategorizationProps) => {
   const categoryLabels = useMemo(
     () =>
-      categorization.elements.map((cat) =>
-        deriveLabelForUISchemaElement(cat, t)
-      ),
-    [categorization, t]
+      filteredCategories.map((cat) => deriveLabelForUISchemaElement(cat, t)),
+    [filteredCategories, t]
   );
 
   return (
     <ul className={subcategoriesClassName}>
-      {categorization.elements.map((category, idx) => {
+      {filteredCategories.map((category, idx) => {
         if (isCategorization(category)) {
           return (
             <li key={categoryLabels[idx]} className={groupClassName}>
               <span>{categoryLabels[idx]}</span>
               <CategorizationList
-                categorization={category}
                 selectedCategory={selectedCategory}
+                filteredCategories={filteredCategories}
```

Comment:
This is causing the endless loop and, thus, the Node out of memory error in the tests.  Basically you hand in the same categories as were handed in the current `CategorizationList`. This results in an endless loop. Instead, you need the elements contained in `category`